### PR TITLE
Issue #30 New rule added; additional modifications

### PR DESCRIPTION
**Pull request by someC0d3r**
- Renamed "IsObjectInAccess" to "IsSymbolAccessible"
- Added rule 0006
- Updated README.md

@StefanMaron 
File "BusinessCentral.LinterCop.csproj.user" can be removed. I should stop VS from creating automatically files in the project directory.

### CODE REVIEW

File: Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            if (table.TableType != TableTypeKind.Temporary)
+                return;
+
+            foreach (var field in table.PrimaryKey.Fields) {
+                if (field.GetProperty(PropertyKind.AutoIncrement) != null) {
+                    context.ReportDiagnostic(
+                        Diagnostic.Create(
+                            DiagnosticDescriptors.Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable,
+                            field.GetLocation(),
+                            new object[] { GetDeclaration(field), field.Name }));
```
Comment (StefanMaron) [inline Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs:39]: the GetDeclaration is not needed. I also would not give any detailed information here. The location of the diagnostig should be enough. Then you can remove the 3rd parameter of Diagnostig.Create() completely and remove the {0} and {1} from the format string of the diagniostic

Comment (someC0d3r) [reply Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs]: Ok! :)

Comment (someC0d3r) [reply Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs]: @StefanMaron 
Should now be done in 
403c90336208a9c788f662101ee9dcfd73b68ed1

File: Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                return;
+
+            CheckTable(tableTypeSymbol, ref context);
+        }
+
+        private void CheckTable(ITableTypeSymbol table, ref SymbolAnalysisContext context) {
+            if (table.TableType != TableTypeKind.Temporary)
+                return;
+
+            foreach (var field in table.PrimaryKey.Fields) {
```
Comment (StefanMaron) [inline Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs:33]: I would go through all fields, as auto does not only work with PK fields

File: Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                IPropertySymbol propertySymbol = field.GetProperty(PropertyKind.AutoIncrement);
+                if (propertySymbol != null) {
```
Comment (StefanMaron) [inline Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs:35]: Also check the value to be true 😁

File: Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs (hunk 2 above)
Comment (someC0d3r) [reply Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs]: Resolved in df1d4cedce602ed4b5e1834ee2b5c066aee5de1d :)

File: Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs (hunk 3 above)
Comment (someC0d3r) [reply Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs]: Resolved in df1d4cedce602ed4b5e1834ee2b5c066aee5de1d :)

**Pull request merged**